catalog/security/catalog-security-ingestplugin/src/main/java/ddf/catalog/security/ingest/IngestPlugin.java

```diff
@@ -0,0 +1,137 @@
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PreIngestPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.security.permission.CollectionPermission;
+import ddf.security.permission.KeyValueCollectionPermission;
+import ddf.security.permission.KeyValuePermission;
+
+/**
+ * IngestPlugin is a PreIngestPlugin that restricts the create/update/delete operations
+ * on the catalog to a group defined by a set of configurable user attributes.
+ */
+public class IngestPlugin implements PreIngestPlugin {
+
+    private String permissionsString;
+
+    private List<KeyValuePermission> permissions = new ArrayList<>();
+
+    public CreateRequest process(CreateRequest createRequest)
+            throws PluginExecutionException, StopProcessingException {
+        authenticateUser(CollectionPermission.CREATE_ACTION);
+        return createRequest;
+    }
+
+    public UpdateRequest process(UpdateRequest updateRequest)
+            throws PluginExecutionException, StopProcessingException {
+        authenticateUser(CollectionPermission.UPDATE_ACTION);
+        return updateRequest;
+    }
+
+    public DeleteRequest process(DeleteRequest deleteRequest)
+            throws PluginExecutionException, StopProcessingException {
+        authenticateUser(CollectionPermission.DELETE_ACTION);
+        return deleteRequest;
+    }
+
+    /**
+     * Getter used by the framework to populate the configuration ui
+     *
+     * @return
+     */
+    public String getPermissionsString() {
+        return this.permissionsString;
+    }
+
+    /**
+     * Get the KeyValuePermission that have been parsed from the permissions string
+     *
+     * @return
+     */
+    public List<KeyValuePermission> getPermissions() {
+        return Collections.unmodifiableList(permissions);
+    }
+
+    /**
+     * Setter used by the ui to set the permissions/attributes
+     *
+     * @param permString
+     */
+    public void setPermissionsString(String permString) {
+        this.permissionsString = permString;
+        parsePermissionsFromString(this.permissionsString);
+    }
+
+    /**
+     * Authenticates the current Subject against the configured ingest permissions.
+     * If the subject doesn't have the right credentials than a StopProcessingException
+     * will be thrown
+     *
+     * @throws StopProcessingException
+     */
+    private void authenticateUser(String action) throws StopProcessingException {
+
+        Subject subject = SecurityUtils.getSubject();
+        KeyValueCollectionPermission kvcp = new KeyValueCollectionPermission(action, permissions);
+        if (subject == null || !subject.isPermitted(kvcp)) {
+            throw new StopProcessingException(
+                    "User is not authorized to create, update or delete records");
+        }
+    }
+
+    /**
+     * Parses a string representation of permission attributes the
+     * same way as the web policy context manager
```

Comment:
Changed IngestPlugin metatype to have a cardinality > 1 to simplify this. Web Context Policy already has a cardinality > 1 so we can't change its logic